spec/Superdesk/ContentApiSdk/API/Request/RequestParametersSpec.php

```diff
@@ -0,0 +1,185 @@
+        $excludeFieldsInputArray = $this->setExcludeFields($excludeFieldsArray)->getExcludeFields();
+        $excludeFieldsInputString = $this->setExcludeFields($excludeFieldsString)->getExcludeFields();
+
+        $excludeFieldsInputArray->shouldBeArray();
+        $excludeFieldsInputString->shouldBeArray();
+        foreach ($excludeFieldsArray as $field) {
+            $excludeFieldsInputArray->shouldContain($field);
+            $excludeFieldsInputString->shouldContain($field);
+        }
+
+        $this->setExcludeFields(null)->getExcludeFields()->shouldReturn(null);
+    }
+
+    function it_should_return_all_parameters()
+    {
+        $this->setPage(10);
+        $this->setMaxResults(1);
+        $this->setQuery('text');
+
+        $arrayReturn = $this->getAllParameters(false);
+        $arrayReturn->shouldBeArray();
+        $arrayReturn->shouldHaveKeyWithValue('page', 10);
+        $arrayReturn->shouldHaveKeyWithValue('max_results', 1);
+        $arrayReturn->shouldHaveKeyWithValue('q', 'text');
+        $this->getAllParameters(true)->shouldBe('q=text&page=10&max_results=1');
+    }
+
+    function it_should_throw_an_exception_on_invalid_start_date()
+    {
+        $this->shouldThrow('\Superdesk\ContentApiSdk\Exception\InvalidArgumentException')->duringSetStartDate('some invalid date');
+        $this->shouldThrow('\Superdesk\ContentApiSdk\Exception\InvalidArgumentException')->duringSetStartDate(array());
+        $this->shouldThrow('\Superdesk\ContentApiSdk\Exception\InvalidArgumentException')->duringSetStartDate(123456789);
+    }
+
+    function it_should_throw_an_exception_on_invalid_end_date()
+    {
+        $this->shouldThrow('\Superdesk\ContentApiSdk\Exception\InvalidArgumentException')->duringSetEndDate('some invalid date');
+        $this->shouldThrow('\Superdesk\ContentApiSdk\Exception\InvalidArgumentException')->duringSetEndDate(array());
+        $this->shouldThrow('\Superdesk\ContentApiSdk\Exception\InvalidArgumentException')->duringSetEndDate(123456789);
+    }
+
+    function it_should_throw_an_exception_on_invalid_query()
+    {
+        $this->shouldThrow('\Superdesk\ContentApiSdk\Exception\InvalidArgumentException')->duringSetQuery(array());
+        $this->shouldThrow('\Superdesk\ContentApiSdk\Exception\InvalidArgumentException')->duringSetQuery(123456789);
+    }
+
+    function it_should_throw_an_exception_on_invalid_page()
+    {
+        $this->shouldThrow('\Superdesk\ContentApiSdk\Exception\InvalidArgumentException')->duringSetPage(array());
+        $this->shouldThrow('\Superdesk\ContentApiSdk\Exception\InvalidArgumentException')->duringSetPage('some page');
+        $this->shouldThrow('\Superdesk\ContentApiSdk\Exception\InvalidArgumentException')->duringSetPage('seven');
+    }
+
+    function it_should_throw_an_exception_on_invalid_max_results()
+    {
+        $this->shouldThrow('\Superdesk\ContentApiSdk\Exception\InvalidArgumentException')->duringSetPage(array());
```

Comment:
shouldn't that be `duringSetMaxResults` ? because this test is the same as the below one